android/java/org/chromium/chrome/browser/crypto_wallet/fragments/TxFragment.java

```diff
@@ -158,7 +159,8 @@ private void setupView(View view) {
                         if (!success || values.length == 0) {
                             return;
                         }
-                        double value = Utils.fromHexWei(mTxInfo.txData.baseData.gasPrice);
+                        double value =
+                                Utils.fromHexWei(mTxInfo.txData.baseData.gasPrice) * gasLimit;
```

Comment:
Shouldn't these calculations be done in BigInteger and then converted only for display?